### PR TITLE
xwayland: minor fixups for stability

### PR DESCRIPTION
**Pull request by Trimutex**
#### Describe your PR, what does it fix/add?
- Fix lock file permissions
- Ensure sockets are closed safely using inline methods
- Attempt to recreate XWayland instance on crash

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
Any socket closes swapped from close() to closeSocketSafely() were differences found in wlroots's method.

#### Is it ready for merging, or does it need work?
Ready

### CODE REVIEW

File: src/xwayland/XWM.cpp [hunk 1]
```diff
@@ -691,6 +691,8 @@ int CXWM::onEvent(int fd, uint32_t mask) {
         Debug::log(CRIT, "XWayland has yeeten the xwm off?!");
         g_pXWayland->pWM.reset();
         g_pXWayland->pServer.reset();
+        // Attempt to create fresh instance
+        g_pXWayland = std::make_unique<CXWayland>(true);
```
Comment (vaxerski) [inline src/xwayland/XWM.cpp:695]: wont this already happen elsewhere?

Comment (Trimutex) [reply src/xwayland/XWM.cpp:695]: I don't think so as every time my XWayland crashes (a separate thing I haven't reasoned out yet) it requires disabling and re-enabling before it works again. Programs instead just pop up as if XWayland were disabled.

Comment (Trimutex) [reply src/xwayland/XWM.cpp:695]: If you have an area where you think this should be put instead I'm down take a look into it

Comment (vaxerski) [reply src/xwayland/XWM.cpp:695]: ah.. right. I see. This should probably be done in a `doLater`, though. Making it instantly here would be an awkward codepath.

Comment (Trimutex) [reply src/xwayland/XWM.cpp]: Good point, newest commit should resolve that

**Review by vaxerski — APPROVED** [review]
thanks!

**Pull request merged**